packages/plugin-exec/sources/ExecFetcher.ts

```diff
@@ -1,9 +1,40 @@
-import {Fetcher, FetchOptions, MinimalFetchOptions}                    from '@yarnpkg/core';
-import {Locator, MessageName}                                          from '@yarnpkg/core';
-import {execUtils, scriptUtils, structUtils, tgzUtils}                 from '@yarnpkg/core';
-import {Filename, NodeFS, PortablePath, npath, ppath, toFilename, xfs} from '@yarnpkg/fslib';
-
-import {PROTOCOL}                                                      from './constants';
+import {execUtils, scriptUtils, structUtils, tgzUtils}                     from '@yarnpkg/core';
+import {Locator, MessageName}                                              from '@yarnpkg/core';
+import {Fetcher, FetchOptions, MinimalFetchOptions}                        from '@yarnpkg/core';
+import {Filename, PortablePath, npath, ppath, toFilename, xfs, NativePath} from '@yarnpkg/fslib';
+
+import {PROTOCOL}                                                          from './constants';
+import {getGeneratorPath}                                                  from './execUtils';
+
+/**
+ * Contains various useful details about the execution context.
+ */
+export interface ExecEnv {
+  /**
+   * The absolute path of the temporary directory where the script runs. Equal to `process.cwd()`.
+   */
+  tempDir: NativePath;
+  /**
+   * The Locator identifying the generator package.
+   */
+  locator: Locator;
+  /**
+   * The absolute path of the generator file. Equal to `process.argv[1]`.
+   */
+  generatorPath: NativePath;
+  /**
+   * The absolute path of the build log directory.
+   */
+  logDir: NativePath;
+  /**
+   * The absolute path of the build log.
+   */
+  logFile: NativePath;
+  /**
+   * The content of the build log. It's a `getter`, so the value is dynamic.
+   */
+  logs: string;
```

Comment:
What's the use case? Doesn't seem very useful, and we might want to change the log system later on.

Reply:
I misinterpreted your message from Discord (https://discordapp.com/channels/226791405589233664/226793810989023234/695587060995719180), thinking that "exposing the build output" referred to the logs, instead of the build directory😅. I also don't really see a use case. I'll remove the log variables.